sympy/ntheory/factor_.py

```diff
@@ -502,31 +502,80 @@ def perfect_power(n, candidates=None, big=True, factor=True):
     sympy.core.intfunc.integer_nthroot
     sympy.ntheory.primetest.is_square
     """
-    if isinstance(n, Rational) and not n.is_Integer:
-        p, q = n.as_numer_denom()
-        if p is S.One:
-            pp = perfect_power(q)
-            if pp:
-                pp = (n.func(1, pp[0]), pp[1])
-        else:
-            pp = perfect_power(p)
-            if pp:
-                num, e = pp
-                pq = perfect_power(q, [e])
-                if pq:
-                    den, _ = pq
-                    pp = n.func(num, den), e
-        return pp
-
-    n = as_int(n)
+    # negative handling
     if n < 0:
-        pp = perfect_power(-n)
-        if pp:
+        if candidates is None:
+            pp = perfect_power(-n, big=True, factor=factor)
+            if not pp:
+                return False
+
             b, e = pp
-            if e % 2:
+            e2 = e & (-e)
+            b, e = b ** e2, e // e2
+
+            if e <= 1:
+                return False
+
+            if big or isprime(e):
                 return -b, e
+
+            for p in primerange(3, e + 1):
+                if e % p == 0:
+                    return - b ** (e // p), p
+
+        odd_candidates = {i for i in candidates if i % 2}
+        if not odd_candidates:
+            return False
+
+        pp = perfect_power(-n, odd_candidates, big, factor)
+        if pp:
+            return -pp[0], pp[1]
+
         return False
 
+    # non-integer handling
+    if isinstance(n, Rational) and not isinstance(n, Integer):
+        p, q = n.p, n.q
+
+        if p == 1:
+            qq = perfect_power(q, candidates, big, factor)
+            return (S.One / qq[0], qq[1]) if qq is not None else False
+
+        if not all(nd := (perfect_power(i, factor=factor) for i in (p, q))):
+                return False
+        (num_base, num_exp), (den_base, den_exp) = nd
```

Comment:
```suggestion
```suggestion
        if not (pp:=perfect_power(p, factor=factor)):
                return False
        if not (qq:=perfect_power(q, factor=factor)):
                return False
        (num_base, num_exp), (den_base, den_exp) = pp, qq
```